### PR TITLE
Add support for 'image' volume driver

### PR DESCRIPTION
**Pull request by mheon**
We added the concept of image volumes in 2.2.0, to support inspecting an image from within a container. However, this is a strictly read-only mount, with no modification allowed.

By contrast, the new `image` volume driver creates a c/storage container as its underlying storage, so we have a read/write layer. This, in and of itself, is not especially interesting, but what it will enable in the future is. If we add a new command to allow these image volumes to be committed, we can now distribute volumes - and changes to them - via a standard OCI image registry (which is rather new and quite exciting).

Future work in this area:
- Add support for `podman volume push` (commit volume changes and push resulting image to OCI registry).
- Add support for `podman volume pull` (currently, we require that the image a volume is created from be already pulled; it would be simpler if we had a dedicated command that did the pull and made a volume from it)
- Add support for scratch images (make an empty image on demand to use as the base of the volume)
- Add UOR support to `podman volume push` and `podman volume pull` to enable both with non-image volume drivers

```release-note
Added support for a new volume driver, `image`, which allows volumes to be created that are backed by images.
```

### CODE REVIEW

**Comment by mheon** [comment]
Still needs manpage updates and tests.

@vrothberg Thanks for the inspiration here, and PTAL when you get a chance

**Comment by rhatdan** [comment]
So this is going to create a layered volume, just not necessarily an executable image?  Interesting idea.
Would you have a podman volume commit with a tag?

**Comment by mheon** [comment]
Probably? Still need to work out the UI bits of how `volume push` and `volume pull` ought to work. Fortunately we have meetings to discuss it at.

**Comment by mheon** [comment]
Manpages updated, test added. This should be ready. @containers/podman-maintainers PTAL

**Review by Luap99 — COMMENTED** [review]
I see one big problem here, this is not backwards compatible. What if I called my volume driver `image` in containers.conf. I think this needs to be rewritten so this can still work if this name is used in containers.conf

**Comment by mheon** [comment]
So prefer a volume plugin named `image` over the volume driver named `image`?

File: test/e2e/volume_create_test.go [hunk 1]
```diff
@@ -162,4 +162,58 @@ var _ = Describe("Podman volume create", func() {
...
+		Expect(psCmd.OutputToString()).To(BeEmpty())
+
+		volumesCmd := podmanTest.Podman([]string{"volume", "ls", "-q"})
+		volumesCmd.WaitWithDefaultTimeout()
+		Expect(volumesCmd).Should(Exit(0))
+		Expect(volumesCmd.OutputToString()).To(Not(ContainSubstring(volName)))
+	})
+
+	It("image-backed volume force removal", func() {
+		podmanTest.AddImageToRWStore(fedoraMinimal)
```
Comment (Luap99) [inline test/e2e/volume_create_test.go:194]: What I want is an actual image plugin name test so we know that we do not break that. I guess renaming one of the existing plugin tests should work.

Comment (mheon) [reply test/e2e/volume_create_test.go:194]: Ack, I can add one.

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by mheon** [comment]
Cirrus appears to be very angry.

**Comment by mheon** [comment]
CI seems to be passing now

**Comment by mheon** [comment]
This is ready.

File: docs/source/markdown/podman-volume-create.1.md [hunk 2]
```diff
@@ -17,7 +17,13 @@ driver options can be set using the **--opt** flag.
 
 #### **--driver**=*driver*
 
-Specify the volume driver name (default **local**). Setting this to a value other than **local** Podman attempts to create the volume using a volume plugin with the given name. Such plugins must be defined in the **volume_plugins** section of the **[containers.conf(5)](https://github.com/containers/common/blob/main/docs/containers.conf.5.md)** configuration file.
+Specify the volume driver name (default **local**).
+There are two drivers supported by Podman itself: **local** and **image**.
```
Comment (rhatdan) [inline docs/source/markdown/podman-volume-create.1.md:21]: Do we support volume plugins?

**Comment by rhatdan** [comment]
/lgtm
Do I see these images with `podman images` command?

**Pull request merged**